### PR TITLE
[For v0.19] Gdml

### PR DESCRIPTION
**Pull request by jimbutter10**
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:

- [ ] Branch rebased on latest master `git pull --rebase upstream master`
- [ ] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- [ ] Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

And please remember to update the Wiki with the features added or changed once this PR is merged.

---

### CODE REVIEW

**Comment by yorikvanhavre** [comment]
There are some errors related to python3 (see the travis log), and also many headers have missing copyright info, which will give us a lot of trouble with linux distributions and might even get FreeCAD kicked out. Also this is a huge PR to handle without any prior discussion. So in any case I rhink this should be looked at after the 0.18 release...

**Comment by luzpaz** [comment]
@jimbutter10 maybe we can make an appimage for it and open it for testing? Lets discuss this in the forums. Please link a discussion thread link to this PR so interested folks can participate. Thanks!

**Comment by wwmayer** [comment]
According to Debian it's considered as non-free and GPL-incompatible:
https://wiki.debian.org/DebianScience/Geant4

Years ago someone asked for help in our forum:
https://forum.freecadweb.org/viewtopic.php?f=10&t=2637&p=19822
http://cad-gdml.in2p3.fr/#developers
https://forum.freecadweb.org/viewtopic.php?f=10&t=2864&p=21980

https://forum.freecadweb.org/viewtopic.php?f=3&t=4147&p=32487p32487#p32487
http://csrsrv1.fynu.ucl.ac.be/csr_web/geant/step-gdml.php

All in all I think it would be problematic to merge this PR and since it's quite unclear what version of Geant4 is used there will be no maintenance any more.

**Comment by galou** [comment]
`*.tex` files should not be in `.gitignore` as they are generally source files. If they are generated files in your case, please filter them out better.

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # file types to ignore
-
+*.html
```
Comment (yorikvanhavre) [inline .gitignore]: html files should not be ignored as there are some currently in the source code

**Comment by yorikvanhavre** [comment]
Also, copyright headers should be fixed:
` *   Copyright (c) YEAR YOUR NAME         <Your e-mail address>            *`

**Comment by luzpaz** [comment]
Please install `codespell` from https://github.com/codespell-project/codespell#updating  
and then run the following command on this branch:  
```
codespell -q 3 -S ./.git,*.po,*.ts -L aci,aline,alle,alledges,als,ang,ba,beginn,behaviour,bloaded,calculater,cancelled,cancelling,cas,cascade,centimetre,childs,colour,colours,currenty,doubleclick,dum,eiter,elemente,feld,freez,iff,indicies,initialisation,initialise,initialised,initialises,initialisiert,ist,kilometre,lod,mantatory,methode,metres,millimetre,modell,nd,noe,normale,normaly,nto,numer,oder,ot,pres,programm,que,recurrance,rougly,seperator,serie,sinc,strack,substraction,te,thist,thru,tread,uint,unter,vertexes,whitespaces ./src/Mod/GDML
```

**Comment by yorikvanhavre** [comment]
There is another license problem with this PR.. It includes the CLHEP library which is GPL3 https://gitlab.cern.ch/CLHEP/CLHEP/blob/master/COPYING

I think since we all agree this PR is quite problematic, and we don't have feedback from its author. (Also apparently the PR author is not the author of the code? It's hosted on Keith Sloan's github). I'm closing it now since none of us has intention to work on this to make it adequate (if possible at all). 

Dont hesitate to reopen it if needed

**Pull request closed without merging**